### ISSUE TITLE
False positive IDE0022, suggested fix breaks the build

### ISSUE THREAD

**Issue by lcsondes** [dotnet/roslyn#80400]
**Version Used**: 

.NET 10 RC 1

**Steps to Reproduce**:

.csproj
```xml
<Project Sdk="Microsoft.NET.Sdk" />
```

.editorconfig to get IDE0022 errors
```
root = true

[*.cs]
dotnet_analyzer_diagnostic.severity = error
csharp_style_expression_bodied_methods = true
```

Class1.cs
```cs
using System.Diagnostics;

public class Class1
{
#if DEBUG
    private int _value;
#endif

    [Conditional("DEBUG")]
    public void AssertValue(int value)
    {
#if DEBUG
        Debug.Assert(_value == value);
#endif
    }
}
```

**Diagnostic Id**:

IDE0022: Use expression body for method

**Expected Behavior**:

No IDE0022

**Actual Behavior**:

IDE0022 with a suggested fix that breaks compilation in Release:

```cs
    [Conditional("DEBUG")]
    public void AssertValue(int value) =>
#if DEBUG
        Debug.Assert(_value == value);
#endif
```

**Comment by MihuBot** [comment]
I'm a bot. Here are possible related and/or duplicate issues (I may be wrong):
- https://github.com/dotnet/roslyn/issues/69783
- https://github.com/dotnet/roslyn/issues/17120

**Comment by CyrusNajmabadi** [comment]
Thanks for the report!
